src/Microsoft.Dism.Tests/DismSessionTest.cs

```diff
@@ -0,0 +1,108 @@
+    {
+        public DismSessionTest(TestWimTemplate template)
+            : base(template)
+        {
+        }
+
+        [Fact]
+        public void SessionOptionsDefaults()
+        {
+            DismSessionOptions options = new DismSessionOptions();
+
+            options.ThrowExceptionOnRebootRequired.ShouldBeTrue();
+        }
+
+        [Theory]
+        [ClassData(typeof(SessionOptionsBehaviorData))]
+        public void SessionOptionsBehavior(DismSessionOptions options, Func<DismSessionOptions, DismSession> sessionFunc)
+        {
+            DismApi.Initialize(DismLogLevel.LogErrors);
+
+            using (DismSession session = sessionFunc(options))
+            {
+                session.Options.ShouldNotBeNull();
+                session.RebootRequired.ShouldBeFalse();
+
+                if (options == null)
+                {
+                    session.Options.ThrowExceptionOnRebootRequired.ShouldBeTrue();
+                }
+                else
+                {
+                    session.Options.ShouldBe(options);
+                }
+
+                DismUtilities.ThrowIfFail(DismApi.ERROR_SUCCESS, session);
+                session.RebootRequired.ShouldBeFalse();
+
+                if (session.Options.ThrowExceptionOnRebootRequired)
+                {
+                    Should.Throw<DismRebootRequiredException>(() => DismUtilities.ThrowIfFail(DismApi.ERROR_SUCCESS_REBOOT_REQUIRED, session));
+                }
+                else
+                {
+                    DismUtilities.ThrowIfFail(DismApi.ERROR_SUCCESS_REBOOT_REQUIRED, session);
+                }
+
+                session.RebootRequired.ShouldBeTrue();
+
+                session.RebootRequired = false;
+                session.RebootRequired.ShouldBeTrue();
+            }
+
+            DismApi.Shutdown();
+        }
+
+        private class SessionOptionsBehaviorData : TheoryData<DismSessionOptions, Func<DismSessionOptions, DismSession>>
+        {
+            private readonly List<DismSessionOptions> _sessionOptions = new List<DismSessionOptions>
+            {
+                null,
+                new DismSessionOptions(),
+                new DismSessionOptions { ThrowExceptionOnRebootRequired = false },
+            };
+
+            private readonly List<Func<DismSessionOptions, DismSession>> _publicSessionMethods = new List<Func<DismSessionOptions, DismSession>>
+            {
+                (_) => DismApi.OpenOnlineSession(),
+                (_) => DismApi.OpenOfflineSession(DismApi.DISM_ONLINE_IMAGE),
+                (_) => DismApi.OpenOfflineSession(DismApi.DISM_ONLINE_IMAGE, null, null),
+            };
+
+            private readonly List<Func<DismSessionOptions, DismSession>> _publicSessionExMethods = new List<Func<DismSessionOptions, DismSession>>
+            {
+                (options) => DismApi.OpenOnlineSessionEx(options),
+                (options) => DismApi.OpenOfflineSessionEx(DismApi.DISM_ONLINE_IMAGE, options),
```

Comment:
This is technically not opening a session to an offline image, I'm not sure if that matters.  You could modify the test to mount an image and then get the offline session, but again, not sure if that matters.  What do you think?

Reply:
Yeah, I sort of gamed the system there. My thought was to make it as quick as possible while still going through the public methods so I could test the new session options are applied (or aren't). So, this was intentional...does is make sense?

Reply:
Yeah, its still testing the code path so let's just leave it as is